USHIFT-1240: add runner for robot framework tests

dhellmann opened the pull request:
I could not figure out how to make the Makefile let me run one test at
a time with valid arguments, so I moved most of the content to a wrapper
script to use instead.

dhellmann commented:
/assign @pacevedom

ggiguash commented on test/variables.yaml.example:
```diff
@@ -3,4 +3,5 @@ USHIFT_HOST: microshift
 USHIFT_USER: microshift
 # SSH Private key to use when logging into MicroShift's host.
+# Unset this variable to use ssh agent.
 SSH_PRIV_KEY: /home/microshift/.ssh/id_rsa
```
Can we use env variable like `$HOME` or `~/.ssh` here instead of `/home/microshift`?

ggiguash commented on test/run.sh:
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
```
We usually add at least `set -e` in the scripts to prevent uncontrolled command failure.

pmtk replied:
I always use opts from http://redsymbol.net/articles/unofficial-bash-strict-mode/ (although IFS might interfere with some things at times)
```
#!/bin/bash
set -euo pipefail
IFS=$'\n\t'
```

dhellmann replied:
done

ggiguash commented on test/run.sh:
```diff
@@ -0,0 +1,84 @@
+    echo "Please create a variables file at ${RF_VARIABLES}" 1>&2
+    echo "See ${RF_VARIABLES}.example for the expected content." 1>&2
+    exit 1
+fi
```
We renamed the default `yaml`, so one needs to make changes to the project for running scripts. 
Can we avoid that by having the defaults in place for running in `devenv`?

dhellmann replied:
I added some documentation explaining how to manage the setup

ggiguash commented on test/run.sh:
```diff
@@ -0,0 +1,84 @@
+
+if [ ! -d "${RF_VENV}" ]; then
+    python3 -m venv "${RF_VENV}" && \
+        "${RF_VENV}/bin/python3" -m pip install -r "${SCRIPTDIR}/requirements.txt"
```
`&&` would not be necessary if `set -e` is there.

ggiguash commented on test/run.sh:
```diff
@@ -0,0 +1,84 @@
+        "${RF_VENV}/bin/python3" -m pip install -r "${SCRIPTDIR}/requirements.txt"
+fi
+
+cd "${SCRIPTDIR}" || (echo "Did not find ${SCRIPTDIR}" 1>&2; exit 1)
```
`set -e` would not require the error message.

ggiguash commented:
/assign @ggiguash

dhellmann reviewed (commented):
Updated based on comments and to add some changes that enable ssh-agent logins for VMs for cases where the ssh key requires a password.

pacevedom commented on test/run.sh:
```diff
@@ -0,0 +1,87 @@
+        --randomize all \
+        --loglevel DEBUG \
+        -V "${RF_VARIABLES}" \
+        -x xunit.xml \
```
CI looks for this file as `junit.xml`, `xunit.xml` wont render in the prow summary.

dhellmann commented:
@pacevedom I added the top-level Makefile target, as we discussed. I removed the Makefile in the test directory as part of that.

@ggiguash I added some setup instructions to explain the configuration file. Let me know if that addresses your concerns about the defaults.

ggiguash commented:
This is LGTM for me. I'm delegating the final approval to @pacevedom

pacevedom commented:
/override ci/prow/microshift-e2e-arm
/override 
ci/prow/e2e-openshift-conformance-reduced-arm
Overriding both of these jobs since arm is flaky in AWS until we figure it out. Failures include unexpected restarts which render connection timeouts and refusals.

pacevedom commented:
/override
ci/prow/e2e-openshift-conformance-reduced-arm

pacevedom commented:
/override ci/prow/e2e-openshift-conformance-reduced-arm

pacevedom reviewed (commented):
/lgtm
